src/connection.ts

```diff
@@ -1369,6 +1369,9 @@ class Connection extends EventEmitter {
       this.messageIo.on('data', (data) => { this.dispatchEvent('data', data); });
       this.messageIo.on('message', () => { this.dispatchEvent('message'); });
       this.messageIo.on('secure', (cleartext) => { this.emit('secure', cleartext); });
+      this.messageIo.on('error', (message) => {
+        this.emit('error', message);
+      });
```

Comment:
I think we need to also dispatch a `socketError` event here so the connection gets cleaned up properly? 🤔 

```suggestion
      this.messageIo.on('error', (error) => {
        this.emit('error', error);
        this.dispatchEvent('socketError', error);
      });
```